### ISSUE TITLE
Disable video recording on cypress run

### ISSUE THREAD

**Issue by ianwalter** [cypress-io/cypress#867]
### Is this a Feature or Bug?

Feature

### Current behavior:

Video recorded on every `cypress run`.

### Desired behavior:

CLI and/or config option to disable video recording on `cypress run`.

### How to reproduce:

Run `cypress run`. See "Started video recording" message and generated video file in `cypress/videos`.

### Additional Info (images, stack traces, etc)

I'm not using the videos while developing and I'd rather save the time/processing and lower my chances of running into associated bugs by disabling this functionality altogether when I need to run all tests.

**Comment by MarcLoupias** [comment]
You can already disable video recording with proper configuration in your cypress.json file ([doc](https://docs.cypress.io/guides/references/configuration.html#Videos)) : `"videoRecording": false`.

**Comment by ianwalter** [comment]
@MarcLoupias Thanks! Not sure how I missed that.

**Comment by PascalPrecht** [comment]
Thanks everyone for this issue as I ran into it as well. However, I do have some thoughts here on usability (but maybe my expectations are wrong):

- I read in the [CLI docs](https://docs.cypress.io/guides/guides/command-line.html#cypress-run) that `--record` specifies "whether to record the test run". With that in mind, I'd expect it to **not** record the test run with `--record` is not passed as an option, or, at least let me disable it a la `cypress run --record false`.

Doing it via config override totally does the trick, it's just less comfortable and to be honest, I didn't run into the overriding docs when searching for such a feature. Frankly I didn't actually search for "overriding config" but rather "how do disable video recording in `cypress run`". 

Again, my expectations may be wrong here. I'm just leaving it here as feedback :)

**Comment by brian-mann** [comment]
@PascalPrecht There's another issue open that talks about how "recording" and "video recording" is confusing nomenclature.

What it's really saying is two different things. `--record` is about recording the run details **on our Dashboard Service** to which you'd have to setup your project to record and explicitly opt in.

By default Cypress takes a video of the run without any flags -- thats the video recording feature. You can turn that off with configuration or CLI flags. `--config videoRecording=false`

**Comment by bahmutov** [comment]
The `--record` is for recording test results _on the dashboard_. Which is separate from video recording for example. The way I usually set this up (because I don't want to record video locally when testing)

- disable video recording in the cypress.json
- use simple `test: cypress run` script in the package.json file
- on CI do `npm test -- --record` and set `CYPRESS_RECORD_KEY=<private record key>` environment variable _and_ `CYPRESS_VIDEO_RECORDING=1` to enable video recording on CI

**Comment by bahmutov** [comment]
hahaha @brian-mann we posted together!

**Comment by brian-mann** [comment]
@PascalPrecht here's the other issue for reference https://github.com/cypress-io/cypress/issues/562

**Comment by PascalPrecht** [comment]
Thanks for the clarification! This all makes total sense. I assume tho then, that this has potential for improvement. Not sure if I'm the only one who got confused by this.

**Comment by johannordin** [comment]
`videoRecording` is now renamed to `video` (if someone stumbles upon this issue)

**Comment by aries1980** [comment]
For the “record”, it is still videoRecording in 3.0.1 .

**Comment by jennifer-shehane** [comment]
Hey @aries1980, what key are you referring to? Where are you using the `videoRecording` key in 3.0.1?

If you were to use the `videoRecording` key in your configuration in Cypress version 3.0+, you should see this error:

```
A configuration option you have supplied has been renamed.

Please rename videoRecording to video
```

**Comment by kud** [comment]
It's still `videoRecording` in the doc: https://docs.cypress.io/guides/core-concepts/screenshots-and-videos.html#Videos

**Comment by jennifer-shehane** [comment]
I created a new issue in our [docs](https://github.com/cypress-io/cypress-documentation) to document the incorrect key reference here: https://github.com/cypress-io/cypress-documentation/issues/1030. Our documentation is open source and contributions are welcome. 😄

**Comment by kud** [comment]
Thank you. @jennifer-shehane :)

**Comment by johnjensenish** [comment]
```
cypress run --config video=false
```

**Comment by KhizerRehan** [comment]
@jennifer-shehane, @MarcLoupias lo is there any possibilty to record video for only failed specs not for passed test cases and used them in report if any test suite fails due to any test case. Thanks. I hope you got my point.

**Comment by jennifer-shehane** [comment]
@KhizerRehan There is no way to only record video for 'failed' specs because we don't know what will fail or pass until the tests are over and the recording is done. You can delete the videos for passed specs though with this code: https://on.cypress.io/after-spec-api#Delete-the-recorded-video-if-the-spec-passed

**Comment by jamesopti** [comment]
> @KhizerRehan There is no way to only record video for 'failed' specs because we don't know what will fail or pass until the tests are over and the recording is done. You can delete the videos for passed specs though with this code: https://on.cypress.io/after-spec-api#Delete-the-recorded-video-if-the-spec-passed

I believe what others and myself are looking for is a way to opt out of sending _anything_ to the dashboard in the event that all tests pass, as this is what counts against the dashboard usage quotas.

I have absolutely no use case for reviewing passed/successful runs on the dashboard, they just burn through the dashboard allocation/quota.

**Comment by rnike** [comment]
@jennifer-shehane  Is it possible to rerun a test case with video recording on failure? If the test case is successful we can just pass it without recording, if possible this will definite fast up our CI.

**Comment by salazarm** [comment]
> @KhizerRehan There is no way to only record video for 'failed' specs because we don't know what will fail or pass until the tests are over and the recording is done. You can delete the videos for passed specs though with this code: https://on.cypress.io/after-spec-api#Delete-the-recorded-video-if-the-spec-passed

If a test fails, you could run the test again and record it that time.

**Comment by hilmanski** [comment]
I write a blog post about this https://onebite.dev/disable-video-recording-when-running-cypress/ with other information regarding video recording

TLDR;
```
import { defineConfig } from 'cypress'

export default defineConfig({
    e2e: {
        setupNodeEvents(on, config) {
            // implement node event listeners here
        },
    },
    video: false,
})
```
in your cypress setting

**Comment by kleinfreund** [comment]
I found that `cypress run --config video=true` doesn’t turn on the video setting if it’s turned off in the config. What does seem to work is using `CYPRESS_video=true`:

```sh
CYPRESS_video=true cypress run
```
